### ISSUE TITLE
Units getting permanently stuck on drawbridges

### ISSUE THREAD

**Issue by Heroesflorian** [Sh0wdown/UnofficialCrusaderPatch#212]
Every now and then it can be seen on AI battles that units don't just stand on a drawbridge with closed gate temporarily, preventing it from getting raised, but actually get stuck on the drawbridges permanently.

In particular, units can get stuck on the corner tiles of a drawbridge, as those are accessible/traversable when the drawbridge is open - but inaccessible (like rocks) when the drawbridge is closed.
So, ~probably if units happen to enter one of the corner tiles right when the drawbridge starts lifting (switching corner accessibility state)~, they are trapped there, and usually stay there "forever" / until they die, ~even when the drawbridge opens and closed again later~ causing the drawbridge to remain open potentially indefinitely.

~(And I actually wouldn't be surprised if the state switch of those corner tiles between accessible and inaccessible was also the reason for - or at least related to - the "demolish all pseudo-unreachable buildings" bug.)~

Okay, I just checked ingame and it seems the corner tiles are inaccessible always... hmm... I really had thought they were not... well, guess you never stop learning. Anyway, this means the "units getting trapped" does instead happen when a drawbridge is **built** while a unit stands/walks on the future drawbridge corner tiles. 
This is slightly less bad as it is a less frequent event, but nonetheless it is just frustrating... 

![image](https://user-images.githubusercontent.com/12549394/51000978-de135a80-152e-11e9-9052-57d1a803ba06.png)

... I proudly present to you the true, one and only **evil chicken**! As long as the **evil chicken** is alive, the drawbridge will not be raised again.

**Comment by Sh0wdown** [comment]
But can't this happen with any other building as well? I don't know how this could be fixed.

**Comment by Monsterfisch** [comment]
With other buildings, the units simply despawn when the computer builds over them instead of them getting stuck inside the building ^^ arguably its not a game breaking issue anyways what you could try is have the drawbridge close even when people are on top of it dropping them in the water?^^ its not like they could go through the gate anyways when the door closes. even though the way it now creates some more unpredictable results that make the game more interesting overall.

**Comment by Heroesflorian** [comment]
Indeed... for "general" buildings, units just despawn. And even within the empty areas of dairy or apple farms, units despawn when the building is placed by the AI, even though units can stand in those areas after the building was placed... 
For towers, the corner tiles are inaccessible, and only units standing there despawn when towers get placed, while units on the other tower tiles "survive". 
For drawbridges, on the other hand, there is no despawning on the inaccessible corner tiles.
